sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.1/async_samples/sample_convert_to_and_from_dict_async.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_convert_to_and_from_dict_async.py
+
+DESCRIPTION:
+    This sample demonstrates how to convert models returned from a recognize operation
+    to and from a dictionary. The dictionary in this sample is then converted to a
+    JSON file, then the same dictionary is converted back to its original model.
+
+USAGE:
+    python sample_convert_to_and_from_dict_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_FORM_RECOGNIZER_ENDPOINT - the endpoint to your Cognitive Services resource.
+    2) AZURE_FORM_RECOGNIZER_KEY - your Form Recognizer API key
+"""
+
+import os
+import json
+import asyncio
+
+async def convert_to_and_from_dict_async():
+    path_to_sample_forms = os.path.abspath(
+        os.path.join(
+            os.path.abspath(__file__),
+            "..",
+            "..",
+            "..",
+            "./sample_forms/id_documents/license.jpg",
+        )
+    )
+
+    from azure.core.serialization import AzureJSONEncoder
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.formrecognizer.aio import FormRecognizerClient
+    from azure.ai.formrecognizer import RecognizedForm
+
+    endpoint = os.environ["AZURE_FORM_RECOGNIZER_ENDPOINT"]
+    key = os.environ["AZURE_FORM_RECOGNIZER_KEY"]
+
+    form_recognizer_client = FormRecognizerClient(
+        endpoint=endpoint, credential=AzureKeyCredential(key)
+    )
+    async with form_recognizer_client:
+        with open(path_to_sample_forms, "rb") as f:
+            poller = await form_recognizer_client.begin_recognize_identity_documents(identity_document=f)
+        
+        id_documents = await poller.result()
+
+    # convert the received model to a dictionary
+    recognized_form_dict = [doc.to_dict() for doc in id_documents]
+
+    # save the dictionary as a JSON content in a JSON file
```

Comment:
maybe leave a comment here that AzureJSONEncoder is only available in azure.core >= 1.18.0? I think our min dep is a bit lower for FR.

Reply:
done!